oversweet-api/src/main/java/com/depromeet/oversweet/drink/vo/DrinkStatisticsTotalInfo.java

```diff
@@ -1,23 +1,51 @@
 package com.depromeet.oversweet.drink.vo;
 
 import com.depromeet.oversweet.domain.record.entity.RecordEntity;
-import com.depromeet.oversweet.drink.dto.DrinkDailySugarTotalStatisticsInfo;
 
 import java.util.List;
+import java.util.stream.Collectors;
 
-/*
+/**
     RecordEntity를 이용해서 총합 (당, 칼로리, 남은 당)을 계산하기 위한 클래스입니다.
  */
 public record DrinkStatisticsTotalInfo(
         List<RecordEntity> recordEntities
 ){
-    public DrinkDailySugarTotalStatisticsInfo getTotalStatisticsInfo(final int dailySugar){
+    public DrinkDailySugarTotalStatisticsInfo getDailyTotalStatisticsInfo(final int dailySugar){
         final int totalIntakeSugar = getTotalIntakeSugar();
         final int totalCalorie = getTotalCalorie();
         final int remainingSugar = getRemainingSugar(totalIntakeSugar, dailySugar);
 
         return DrinkDailySugarTotalStatisticsInfo.of(dailySugar, remainingSugar, totalIntakeSugar, totalCalorie);
     }
+
+    public DrinkWeeklySugarTotalStatisticsInfo getWeeklyTotalStatisticsInfo(){
+        final int totalIntakeSugar = getTotalIntakeSugar();
+        final int dailyAverageSugar = getDailyAverageSugar();
+        final int totalCalorie = getTotalCalorie();
+        return new DrinkWeeklySugarTotalStatisticsInfo(totalIntakeSugar, dailyAverageSugar, totalCalorie);
+    }
+
+    public List<DrinkDailySugarInfo> getDayOfWeekSugarInfo() {
+        return recordEntities.stream()
+                .collect(Collectors.groupingBy(record -> record.getCreatedAt().toLocalDate().atStartOfDay()))
+                .entrySet()
+                .stream()
+                .map(entry -> DrinkDailySugarInfo.of(entry.getKey(), getTotalIntakeSugar(entry.getValue())))
+                .toList();
+    }
+
+    private int getDailyAverageSugar() {
+        final int totalIntakeSugar = getTotalIntakeSugar();
+        return totalIntakeSugar / recordEntities.size();
+    }
```

Comment:
1주일간 총 섭취한 당류을 구하고, 일별 평균 당 섭취량을 구하는 목적인데
recordEntities.size() 로 나누는건 아니지 않을까요??
제 생각은 7일로 나눠야 할 듯해요


```suggestion
    private int getDailyAverageSugar() {
        final int totalIntakeSugar = getTotalIntakeSugar();
        return totalIntakeSugar / SEVEN_DAYS;
    }
```

Reply:
이 부분에 대해서 고민이 많았는데 만약 유저가 7일이 아니라 3일만 먹고 그에대한 통계를 볼 수 도 있다고 생각하는데 어떻게 생각하세요? 3일만 먹었다고 해서 주간 통계가 아예 다 안보이면 이상할 것 같아서요

Reply:
그렇다면 위에서 언급했던 summaryStatistics 을 활용하면 좋을 것 같습니다